crates/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -205,6 +207,15 @@ impl From<TypeCheckError> for Diagnostic {
 
                 Diagnostic::simple_error(message, String::new(), span)
             }
+            TypeCheckError::CallDeprecated { span, ref note, .. } => {
+                let mut message = error.to_string();
+
+                if let Some(note) = note {
+                    message += &format!(": {note}");
+                }
+
+                Diagnostic::simple_warning(message, String::new(), span)
```

Comment:
I noticed the secondary message is empty. What do you think of using the secondary message for the note? I'm not sure what it looks like (or what rust's deprecated warnings look like) off the top of my head so I'll defer to which you think looks better.